lib/migrator.js

```diff
@@ -1,180 +1,270 @@
+const { transform } = require("./rewrite-imports");
 
 module.exports = class Migrator {
   constructor(options) {
     const { projectRoot, structure } = options;
 
     this.projectRoot = projectRoot;
     this.structure = structure;
-  }
-
-  findClassicComponentTemplates() {
-    const templateFolderPath = path.join(this.projectRoot, 'app/templates/components');
-    const templateFilePaths = glob.sync(`${templateFolderPath}/**/*.hbs`);
 
-    return templateFilePaths;
-  }
-
-  findClassicComponentClasses() {
-    const classFolderPath = path.join(this.projectRoot, 'app/components');
-    const classFilePaths = glob.sync(`${classFolderPath}/**/*.{js,ts}`);
-
-    return classFilePaths;
-  }
-
-  findTemplates() {
-    const templateFolderPath = path.join(this.projectRoot, 'app/templates');
-    const templateFilePaths = glob.sync(`${templateFolderPath}/**/*.hbs`);
+    const pkg = JSON.parse(
+      fs.readFileSync(path.join(this.projectRoot, "package.json"), {
+        encoding: "utf-8",
+      })
+    );
 
-    return templateFilePaths;
+    this.type = !!pkg["ember-addon"] ? "addon" : "app";
   }
 
-  skipTemplatesUsedAsLayoutName(templateFilePaths) {
-    console.info(`\nChecking if any component templates are used as templates of other components using \`layoutName\``);
-
+  execute() {
     const classFilePaths = this.findClassicComponentClasses();
-    const componentsWithLayoutName = getLayoutNameTemplates(classFilePaths);
 
-    if (componentsWithLayoutName.length) {
-      componentsWithLayoutName.sort().forEach(component => {
-        console.info(`❌ Did not move '${component}' due to usage as "layoutName" in a component`);
-      });
+    const componentsWithLayoutName = getLayoutNameTemplates(classFilePaths);
+    logSkipped(
+      componentsWithLayoutName,
+      "Checking if any component templates are used as templates of other components using `layoutName`",
+      'usage as "layoutName" in a component'
+    );
 
-      templateFilePaths = templateFilePaths.filter(templateFilePath => {
-        // Extract '/app/templates/components/nested1/nested-component.hbs'
-        const filePathFromApp = templateFilePath.slice(this.projectRoot.length);
+    const componentsWithPartial = getPartialTemplates(this.findTemplates());
+    logSkipped(
+      componentsWithPartial,
+      "Checking if any component templates are used as partials",
+      'usage as a "partial"'
+    );
 
-        // Extract '/components/nested1/nested-component.hbs'
-        const filePathFromAppTemplates = filePathFromApp.slice('app/templates/'.length);
+    let templateFilePaths = this.findClassicComponentTemplates()
+      .filter(skipTemplatesUsedAsLayoutName(componentsWithLayoutName, this))
+      .filter(skipTemplatesUsedAsPartial(componentsWithPartial, this));
 
-        // Extract 'components/nested1/nested-component'
-        const classFilePath = filePathFromAppTemplates.slice(1).replace('.hbs', '');
+    templateFilePaths.forEach(transform);
```

Comment:
This is not quite correct; it needs to run for *class* paths associated with a template which is actually being migrated. Getting that should be straightforward, either using the `classFilePaths` information or simply running a `.map` on the `templateFilePaths` to transform `templates/components` to `components` and `.hbs` to `.js`.

Reply:
Resolved via 3ea6f6f